cmd/otel-allocator/target/discovery.go

```diff
@@ -75,6 +79,17 @@ func (m *Discoverer) ApplyConfig(source allocatorWatcher.EventSource, cfg *confi
 		}
 	}
 
+	hash, err := hashstructure.Hash(m.jobToScrapeConfig, nil)
+	if err != nil {
+		return err
+	}
+	// If the hash has changed, updated stored hash and send the new config.
+	// Otherwise skip updating scrape configs.
+	if m.scrapeConfigsUpdater != nil && m.scrapeConfigsHash != hash {
+		m.scrapeConfigsHash = hash
```

Comment:
as mentioned above, this should only come after our successful call to the `UpdateScrapeConfigResponse`